docs/lazy-adr/adr-006-row-propagation.md

```diff
@@ -0,0 +1,56 @@
+# ADR 006: Block Propagation with Rows
+
+## Changelog
+* 24.06.2021
+
+## Context
+It's a long story of relations between Celestia, Tendermint, and consensus block gossiping. Celestia's team discussed multiple ideas, several ADRs were made, and nothing yet was finalized. The ADR is another attempt to bring changes into block gossiping and hopefully successful.
+
+Currently, we inherit the following from Tendermint. Our codebase relies on the blocks Parts notion, which is a piece of an entire serialized block. Those Parts are gossiped between nodes in consensus and committed with PartSetHeader containing a Merkle Root of the Parts. However, Parts gossiping wasn't designed for Celestia blocks.
+
+Celestia comes with a different block representation from Tendermint. It lays out block as a table of data shares, where Rows or Columns can be and should be gossiped instead of Parts, keeping only one system-wide commitment to data.
```

Comment:
it's worth noting that we can remove the PartSetHeader from the BlockId/Header (and hence this commitment to the data) independent of its usage during consensus (or how data is gossiped):

 - https://github.com/tendermint/spec/issues/78
 - https://github.com/tendermint/spec/issues/277